e2e/tests/block/create_block_test.go

```diff
@@ -0,0 +1,177 @@
+package block
+
+import (
+	"fmt"
+	"github.com/rook/rook/e2e/framework/clients"
+	"github.com/rook/rook/e2e/framework/contracts"
+	"github.com/rook/rook/e2e/framework/enums"
+	"github.com/rook/rook/e2e/framework/manager"
+	"github.com/rook/rook/e2e/framework/objects"
+	"github.com/rook/rook/pkg/model"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	"strings"
+	"testing"
+)
+
+// Test Common Block Image Creation Scenarios. These tests should be platform independent.Platform specific tests are present
+//in their own package.
+// All Block Images created for testing purpose should have the work "test" in them(so that they can be cleaned up later)
+var (
+	env      objects.EnvironmentManifest
+	platform enums.RookPlatformType
+)
+
+//One init function per package - initializes Rook infra and installs rook(if needed based on flags)
+func init() {
+	env = objects.NewManifest()
+	var err error
+
+	rookPlatform, err := enums.GetRookPlatFormTypeFromString(env.Platform)
+	if err != nil {
+		panic(fmt.Errorf("Cannot get platform", err))
+	}
+
+	k8sVersion, _ := enums.GetK8sVersionFromString(env.K8sVersion)
+
+	rookTag := env.RookTag
+	if err != nil {
+		panic(fmt.Errorf("Rook Tag is required", err))
+	}
+
+	err, rookInfra := rook_test_infra.GetRookTestInfraManager(rookPlatform, true, k8sVersion)
+	if err != nil {
+		panic(fmt.Errorf("Error during Rook Infra Setup", err))
+	}
+	skipRookInstall := env.SkipInstallRook == "true"
+
+	rookInfra.ValidateAndSetupTestPlatform(skipRookInstall)
+
+	err = rookInfra.InstallRook(rookTag, skipRookInstall)
```

Comment:
why not just use an if statement to control whether InstallRook is executed, instead of pushing the logic down into the method....